armi/reactor/converters/uniformMesh.py

```diff
@@ -85,6 +86,317 @@ def converterFactory(globalFluxOptions):
+    """
+    The :py:class:`UniformMeshGenerator <armi.reactor.converters.UniformMeshGenerator>`
+    generates a common axial mesh to for the uniform mesh converter to use. The generation
+    algorithm starts with the simple ``average1DWithinTolerance`` utility function to compute
+    a representative "average" of the assembly meshes in the reactor. It then modifies that
+    mesh to more faithfully represent important material boundaries of fuel and control absorber
+    material.
+
+    The decusping feature is controlled with the case setting ``uniformMeshMinimumSize``. If no
+    value is provided for this setting, the uniform mesh generator will skip the decusping step
+    and just provide the result of ``_computeAverageAxialMesh``.
+    """
+
+    FLOAT_ROUNDING_DECIMALS = 12
+
+    def __init__(self, r, minimumMeshSize=None):
+        """
+        Initialize an object to generate an appropriate common axial mesh to use for uniform mesh conversion.
+
+        Parameters
+        ----------
+        r : :py:class:`Reactor <armi.reactor.reactors.Reactor>` object.
+            Reactor for which a common mesh is generated
+        minimumMeshSize : float, optional
+            Minimum allowed separation between axial mesh points in cm
+            If no miminmum mesh size is provided, no "decusping" is performed
+        """
+        self._sourceReactor = r
+        self.minimumMeshSize = minimumMeshSize
+        self._commonMesh = None
+
+    def generateCommonMesh(self):
+        """
+        Generate a common axial mesh to use.
+
+        Notes
+        -----
+        Attempts to reduce the effect of fuel and control rod absorber smearing
+        ("cusping" effect) by keeping important material boundaries in the
+        common mesh.
+        """
+        self._computeAverageAxialMesh()
+        if self.minimumMeshSize is not None:
+            self._decuspAxialMesh()
+
+    def _computeAverageAxialMesh(self):
+        """
+        Computes an average axial mesh based on the core's reference assembly.
+
+        Notes
+        -----
+        This iterates over all the assemblies in the core and collects all assembly meshes
+        that have the same number of fine-mesh points as the `refAssem` for the core. Based on
+        this, the proposed uniform mesh will be some average of many assemblies in the core.
+        The reason for this is to account for the fact that multiple assemblies (i.e., fuel assemblies)
+        may have a different mesh due to differences in thermal and/or burn-up expansion.
+        """
+        src = self._sourceReactor
+        refAssem = src.core.refAssem
+
+        refNumPoints = len(src.core.findAllAxialMeshPoints([refAssem])) - 1
+        allMeshes = []
+        for a in src.core:
+            # Get the mesh points of the assembly, neglecting the first coordinate
+            # (typically zero).
+            aMesh = src.core.findAllAxialMeshPoints([a])[1:]
+            if len(aMesh) == refNumPoints:
```

Comment:
add a comment calling out that this is a pretty bad check since its intending to check that they have simlar mesh but is only checking that the len() is equal. it could fail pretty spectacularly in corner cases where len is equal but mesh is much different. i know you inhered the code though, so we dont need to fix it.

Reply:
I think it intends to do exactly what it does, which is check whether the two meshes have the same number of points. If the mesh does have the correct number of points, it is added to the list to be averaged together with the other meshes.

The corner case where the mesh is much different is handled by `average1DWithinTolerance`. The tolerance is not a user-configurable setting, but it defaults to 20%. If the mesh size for any block in a given assembly mesh differs from the overall average by more than 20%, that assembly mesh will be removed and the overall average will be recalculated.

Reply:
"If the mesh does have the correct number of points, it is added to the list to be averaged together with the other meshes." 

I think the subtext of this is that its testing "are these meshes similar enough to reasonably average. Length is not sufficient for this. For example if you had reflector meshes within 5%, you wouldn't want tat mesh to drag the uniform mesh, but it would with the current code.

I think there is room for improvement here, eg checking assembly type or something a little more rigorous than equal length. 

not part of this PR though.